### PR TITLE
Further resources & regions

### PR DESCRIPTION
**Pull request by fvanderbiest**
This is a fix for #443 

In addition:
 * the relation between `Region` and `AministrativeDivision` was simplified (no need for an association class)
 * tests have been overhauled

### CODE REVIEW

File: thinkhazard/tests/views/__init__.py [hunk 1]
```diff
@@ -82,89 +79,174 @@ def populate_db():
...
+    admin_div_20.geom = geometry
+    DBSession.add(admin_div_20)
 
     shape = MultiPolygon([
         Polygon([(0, 0), (0, 1), (.5, 1), (.5, 0), (0, 0)])
     ])
     geometry = from_shape(shape, 4326)
 
-    div_level_3_1 = AdministrativeDivision(**{
+    # admin_div_31 is a department (division level 3)
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: I think divisions are either country, province or region (ie. not department).
https://github.com/GFDRR/thinkhazard/blob/master/thinkhazard/scripts/initializedb.py#L101

File: thinkhazard/tests/views/__init__.py [hunk 2]
```diff
@@ -82,89 +79,174 @@ def populate_db():
...
+    # it's parent is admin_div_20
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: its

File: thinkhazard/tests/views/__init__.py [hunk 3]
```diff
@@ -82,89 +79,174 @@ def populate_db():
...
+        'code': 13,
+        'leveltype_id': 1,
+        'name': u'Division level 1 4'
+    })
+    admin_div_13.geom = geometry
+    DBSession.add(admin_div_13)
 
-    div_level_2 = AdministrativeDivision(**{
+    # admin_div_20 is a region (division level 2)
+    # it's parent is admin_div_10
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: its

File: thinkhazard/tests/views/__init__.py [hunk 4]
```diff
@@ -82,89 +79,174 @@ def populate_db():
...
+    admin_div_13 = AdministrativeDivision(**{
+        'code': 13,
+        'leveltype_id': 1,
+        'name': u'Division level 1 4'
+    })
+    admin_div_13.geom = geometry
+    DBSession.add(admin_div_13)
 
-    div_level_2 = AdministrativeDivision(**{
+    # admin_div_20 is a region (division level 2)
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: Province (see comment below).
https://github.com/GFDRR/thinkhazard/blob/master/thinkhazard/scripts/initializedb.py#L101

File: thinkhazard/processing/harvesting.py [hunk 5]
```diff
@@ -209,9 +208,7 @@ def populate_region_administrativedivision_association(dry_run=False):
                 continue
 
             try:
-                association = RegionAdministrativeDivisionAssociation()
-                association.administrativedivision = admindiv
-                region.administrativedivisions.append(association)
+                region.administrativedivisions.append(admindiv)
                 DBSession.add(region)
```
Comment (arnaud-morvan) [inline thinkhazard/processing/harvesting.py]: If region come from session, it is already bind, no need to add it to session.

File: thinkhazard/processing/harvesting.py [hunk 6]
```diff
@@ -240,11 +237,8 @@ def harvest_region(object):
     else:
         logger.info(u'Updating Region - {}'.format(name))
         # drop existing relationships with GAUL administrative divisions
-        assocs = DBSession.query(RegionAdministrativeDivisionAssociation) \
-            .filter(RegionAdministrativeDivisionAssociation
-                    .region_id == id).all()
-        for a in assocs:
-            DBSession.delete(a)
+        for ad in region.administrativedivisions:
+            region.administrativedivisions.remove(ad)
```
Comment (arnaud-morvan) [inline thinkhazard/processing/harvesting.py]: why do not truncate association table directly

File: thinkhazard/tests/views/__init__.py [hunk 7]
```diff
@@ -82,89 +79,174 @@ def populate_db():
...
+        })
+    )
+
+    # admin_div_10 has (EQ, HIGH)
+    admin_div_10.hazardcategories.append(
+        HazardCategoryAdministrativeDivisionAssociation(**{
+            'hazardcategory': category_eq_hig
+        })
+    )
+    # admin_div_11 has no category (this is tested)
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: Please extra space.

File: thinkhazard/tests/views/__init__.py [hunk 8]
```diff
@@ -82,89 +79,174 @@ def populate_db():
...
+    # admin_div_12 has (EQ, HIGH)
+    admin_div_12.hazardcategories.append(
+        HazardCategoryAdministrativeDivisionAssociation(**{
+            'hazardcategory': category_eq_hig
+        })
+    )
+    # admin_div_13 has (EQ, HIGH)
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: Please add extra space.

File: thinkhazard/tests/views/__init__.py [hunk 9]
```diff
@@ -208,19 +290,21 @@ def populate_db():
...
+    admin_div_32.hazardcategories.append(
         HazardCategoryAdministrativeDivisionAssociation(**{
             'hazardcategory': category_fl_med
         })
     )
-    DBSession.add(div_level_3_2)
+    DBSession.add(admin_div_32)
 
+    # generic further resource for EQ:
+    # it should be found on every EQ report page
```
Comment (pgiraud) [inline thinkhazard/tests/views/__init__.py]: Well except `admin_div_12`.

**Comment by pgiraud** [comment]
LGTM.

**Pull request merged**